### PR TITLE
drivers: bluetooth: add stubs to allow CI testing without blobs

### PR DESCRIPTION
**Pull request by sylvioalves**
Add all bluetooth stubs in order to allow building without blobs.

### CODE REVIEW

File: zephyr/esp32/CMakeLists.txt [hunk 1]
```diff
@@ -368,32 +368,34 @@ if(CONFIG_SOC_SERIES_ESP32)
...
-      core
-      pp
-      phy
-      rtc
-      ## esp-idf wifi libs refer gcc libs symbols, so linked in libgcc
-      gcc
-        -L${CMAKE_CURRENT_SOURCE_DIR}/../blobs/lib/esp32
-    )
-
+    if (NOT(CONFIG_WIFI_BUILD_ONLY_MODE OR CONFIG_BT_BUILD_ONLY_MODE))
```
Comment (raffarost) [inline zephyr/esp32/CMakeLists.txt:371]: should it be AND instead of OR?
if only one is in build mode, the lib should be included as well

**Review by raffarost — APPROVED** [review]

File: zephyr/esp32/CMakeLists.txt (hunk 1 above)
Comment (LucasTambor) [reply zephyr/esp32/CMakeLists.txt:371]: Or perhaps use only one Kconfig for both cases, such as `CONFIG_RF_BUILD_ONLY_MODE`

**Review by wmrsouza — APPROVED** [review]

File: zephyr/esp32/CMakeLists.txt (hunk 1 above)
Comment (jhedberg) [reply zephyr/esp32/CMakeLists.txt:371]: Or even more generic, for any feature that may depend on blobs, that that seems like the ultimate purpose that this is being introduced. E.g. something like `CONFIG_BUILD_ONLY_NO_BLOBS`.

Comment (sylvioalves) [reply zephyr/esp32/CMakeLists.txt:371]: @jhedberg please read my previous comment at https://github.com/zephyrproject-rtos/zephyr/pull/80785#issuecomment-2453265212

**Comment by sylvioalves** [comment]
Only rebased. PTAL again anyway.

**Review by jhedberg — APPROVED** [review]

**Review by wmrsouza — APPROVED** [review]

**Review by raffarost — APPROVED** [review]

**Review by marekmatej — APPROVED** [review]

**Pull request merged**